src/main/java/edu/byu/cs/autograder/compile/ModuleIndependenceVerifier.java

```diff
@@ -5,53 +5,66 @@
-        Path clientPath = context.stageRepo().toPath().resolve("client/src/main/java");
-        Set<String> clientClasses = qualifiedNames(clientFiles, clientPath);
+            Set<File> clientFiles = reader.filesMatching(".*client/src/main/java/.*").collect(Collectors.toSet());
+            Set<String> clientPackages = packageNames(clientFiles, reader);
 
-        try {
-            checkImports(context, reader, serverFiles, clientClasses);
-            checkImports(context, reader, clientFiles, serverClasses);
+            checkImports(context, reader, serverFiles, clientPackages);
+            checkImports(context, reader, clientFiles, serverPackages);
         } catch (IOException e) {
             throw new GradingException("Unable to verify module independence", e);
         }
     }
 
-    private Set<String> qualifiedNames(Set<File> files, Path relativeTo) {
-        Set<String> classes = new HashSet<>();
+    private Set<String> packageNames(Set<File> files, StudentCodeReader reader) throws IOException {
+        Set<String> packages = new HashSet<>();
         for(File file : files) {
-            String fileName = relativeTo.relativize(file.toPath()).toString();
-            if(!fileName.contains(".java")) continue;
-            classes.add(fileName.substring(0, fileName.indexOf('.')).replace('/', '.'));
+            if(!file.getName().contains(".java")) continue;
+            List<String> contents = reader.getFileContents(file);
+            if(contents.isEmpty()) continue;
+            String firstLine = contents.getFirst();
+            if(firstLine.startsWith("package")) {
+                packages.add(firstLine.substring(8, firstLine.indexOf(';')));
+            }
         }
-        return classes;
+        return packages;
     }
 
-    private void checkImports(GradingContext context, StudentCodeReader reader, Set<File> files, Set<String> classes)
+    private void checkImports(GradingContext context, StudentCodeReader reader, Set<File> files, Set<String> packages)
             throws IOException {
         for(File file : files) {
             List<String> contents = reader.getFileContents(file);
-            for (String line : contents) {
-                if (line.matches("(?:abstract )?class|interface|record")) break;
-                for (String classImport : classes) {
-                    if (line.matches("import %s;".formatted(classImport))) {
-                        String warning = "File %s imports %s. The client and server modules should be independent"
-                                .formatted(context.stageRepo().toPath().relativize(file.toPath()), classImport);
-                        context.observer().notifyWarning(warning);
-                    }
+            for (int i = 0; i < contents.size(); i++) {
+                String line = contents.get(i).trim();
+                if (line.isEmpty()) continue;
+                if (!(line.startsWith("package") || line.startsWith("import"))) {
```

Comment:
What do you think about using [this regex](https://regex101.com/r/ZWGoTq/4) to determine when we should `break`?